packages/next/build/swc/src/styled_jsx/mod.rs

```diff
@@ -0,0 +1,786 @@
+    let nearest_scope_bindings = self.nearest_scope_bindings.clone();
+    self.nearest_scope_bindings = collect_decls(&func);
+    let func = func.fold_children_with(self);
+    self.nearest_scope_bindings = nearest_scope_bindings;
+    func
+  }
+
+  fn fold_arrow_expr(&mut self, func: ArrowExpr) -> ArrowExpr {
+    let current_bindings = self.nearest_scope_bindings.clone();
+    self.nearest_scope_bindings = collect_decls(&func);
+    let func = func.fold_children_with(self);
+    self.nearest_scope_bindings = current_bindings;
+    func
+  }
+
+  fn fold_module(&mut self, module: Module) -> Module {
+    self.bindings = collect_decls(&module);
+    self.evaluator = Some(Evaluator::new(module.clone(), Marks::new()));
+    self.style_import_name = Some(get_usable_import_specifier(&module.body));
+    module.fold_children_with(self)
+  }
+}
+
+impl StyledJSXTransformer {
+  fn check_children_for_jsx_styles(&mut self, children: &Vec<JSXElementChild>) {
+    let mut styles = vec![];
+    for i in 0..children.len() {
+      if let JSXElementChild::JSXElement(child_el) = &children[i] {
+        if is_styled_jsx(&child_el) {
+          self.file_has_styled_jsx = true;
+          self.has_styled_jsx = true;
+          let expr = get_style_expr(&child_el);
+          let style_info = self.get_jsx_style(expr, is_global(&child_el));
+          styles.insert(0, style_info);
+        }
+      }
+    }
+    if self.has_styled_jsx {
+      let (static_class_name, class_name) =
+        compute_class_names(&styles, self.style_import_name.as_ref().unwrap());
+      self.styles = styles;
+      self.static_class_name = static_class_name;
+      self.class_name = class_name;
+    }
+  }
+
+  fn get_jsx_style(&mut self, expr: &Expr, is_global_jsx_element: bool) -> JSXStyle {
+    let mut hasher = DefaultHasher::new();
+    let css: String;
+    let css_span: Span;
+    let is_dynamic;
+    let mut expressions = vec![];
+    match expr {
+      Expr::Lit(Lit::Str(Str { value, span, .. })) => {
+        hasher.write(value.as_ref().as_bytes());
+        css = value.to_string().clone();
+        css_span = span.clone();
+        is_dynamic = false;
+      }
+      Expr::Tpl(Tpl {
+        exprs,
+        quasis,
+        span,
+      }) => {
+        if exprs.len() == 0 {
+          hasher.write(quasis[0].raw.value.as_bytes());
+          css = quasis[0].raw.value.to_string();
+          css_span = span.clone();
+          is_dynamic = false;
+        } else {
+          expr.clone().hash(&mut hasher);
+          let mut s = String::new();
+          for i in 0..quasis.len() {
+            let placeholder = if i == quasis.len() - 1 {
+              String::new()
+            } else {
+              format!("__styled-jsx-placeholder__{}", i)
+            };
+            s = format!("{}{}{}", s, quasis[i].raw.value, placeholder)
+          }
+          css = String::from(s);
+          css_span = span.clone();
```

Comment:
`.clone()` is not required for types like `Span`, which implements `Copy`.
If you want to get `Span` from `&Span`, you can do `*span`.